packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,25 +5,26 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {Fragment} from 'react';
-
+import React, {
+  useState,
+  useEffect,
+  Fragment,
+  useContext,
+  useRef,
+  useCallback,
+} from 'react';
 import DocusaurusContext from '@docusaurus/context';
 
 import './styles.css';
 
-class Search extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      enabled: true,
-      isExpanded: false,
-    };
-    this.searchBarRef = React.createRef();
-    this.toggleSearchIconClick = this.toggleSearchIconClick.bind(this);
-  }
+const Search = props => {
+  const [isExpanded, setIsExpanded] = useState(false);
```

Comment:
The expanded state is both in this component and in its parent. Duplicating state is rarely a good idea and now we have to always keep them in sync. Since this state is needed outside this component, this component should not own the state. Instead the parent should own it and pass the value into this component as props.